lib/minitest/assertions.rb

```diff
@@ -170,6 +170,7 @@ def assert_empty obj, msg = nil
     # See also: Minitest::Assertions.diff
 
     def assert_equal exp, act, msg = nil
+      raise ArgumentError, "use assert_nil if expecting nil" if exp.nil?
```

Comment:
I'm no longer fine with messing up the counts. Ugh I use assert_equal a LOT. It's even worse on the spec tests because they're not sandboxed the way the assertion tests are. If I switch to `refute_nil`, I'll jimmy the assertion count to make up for that.